### PR TITLE
refactor: extract text/plain as constant

### LINKED ISSUE #16508: [Code] Extract "text/plain" as constant
Maybe we can add `text/plain` in constants file?

_Originally posted by @ShridharGoel in https://github.com/ankidroid/Anki-Android/pull/16393#discussion_r1606052038_

### Checked for duplicates?

- [X] This issue is not a duplicate

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by shrutigitte**
## Purpose / Description
Extracting 'text/plain' as constant and replacing it in the 

## Fixes
* Fixes #16508 

## How Has This Been Tested?
Tested locally and project build successfully

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by david-allison — COMMENTED** [review]
@ShridharGoel Was this what you were expecting from your comment?

I would expect this to be outside the `constants.xml`, and just as a bare string (to remove the `Context` dependency)

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/utils/AssetHelperTest.kt [hunk 1]
```diff
@@ -30,7 +30,7 @@ class AssetHelperTest {
         assertThat(AssetHelper.guessMimeType("test.mjs"), equalTo("text/javascript"))
         assertThat(AssetHelper.guessMimeType("test.json"), equalTo("application/json"))
         assertThat(AssetHelper.guessMimeType("test.css"), equalTo("text/css"))
-        assertThat(AssetHelper.guessMimeType("test.txt"), equalTo("text/plain"))
+        assertThat(AssetHelper.guessMimeType("test.txt"), equalTo(TEXT_PLAIN))
```
Comment (david-allison) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/utils/AssetHelperTest.kt:33]: keep this hardcoded as it's a test

**Review by david-allison — CHANGES_REQUESTED** [review]
In addition to Shridhar's comment

File: AnkiDroid/src/main/java/com/ichi2/anki/AnkiActivity.kt [hunk 2]
```diff
@@ -596,3 +596,6 @@ fun Fragment.requireAnkiActivity(): AnkiActivity {
     return requireActivity() as? AnkiActivity?
         ?: throw java.lang.IllegalStateException("Fragment $this not attached to an AnkiActivity.")
 }
+
+/** Used for mime type or Intent type when sharing text via other applications **/
+const val TEXT_PLAIN = "text/plain"
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/AnkiActivity.kt:601]: As mentioned in Discord: `AssetHelper.kt`

**Review by david-allison — CHANGES_REQUESTED** [review]

**Review by david-allison — APPROVED** [review]
if CI passes, I'm happy

**Review by criticalAY — APPROVED** [review]
LGTM!

**Pull request merged**